feature/mypage/src/main/java/com/mument_android/mypage/fragment/UnregisterFragment.kt

```diff
@@ -0,0 +1,163 @@
+
+        reasonBtnEvent()
+        reasonChooseBtnEvent()
+
+        reasonChooseTouchEvent()
+
+        isAgreeBtnEvent()
+        unregisterFinish()
+
+    }
+
+    //이유 선택 박스 눌렀을 때
+    private fun reasonChooseBtnEvent() {
+        binding.clReason.setOnClickListener {
+            if (binding.rgChooseReason.isEmpty()) {
+                binding.tvChooseReason.let {
+                    it.setText(R.string.unregister_choose_reason)
+                    it.setTextColor(
+                        ContextCompat.getColor(
+                            requireContext(),
+                            R.color.mument_color_gray1
+                        )
+                    )
+                }
+            }
+            binding.clReason.isSelected = !binding.clReason.isSelected
+            myPageViewModel.clickReasonChooseBox()
+            setAnimationReason()
+            hideKeyboard()
+            myPageViewModel.isSelectSixthReason.value = false
+        }
+    }
+
+    // 이유 선택 했을 때
+    private fun reasonBtnEvent() {
+        binding.rgChooseReason.setOnCheckedChangeListener { _, checkedID ->
+            myPageViewModel.clickReasonChooseBox()
+            myPageViewModel.clickReasonChoose()
+            setAnimationReason()
+
+            //이유선택했을 때 이유선택박스 text color 변경
+            binding.tvChooseReason.setTextColor(
+                ContextCompat.getColor(
+                    requireContext(),
+                    R.color.mument_color_black1
+                )
+            )
+            binding.clReason.isSelected = false
+
+            when (checkedID) {
+                R.id.unregister_reason_first -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_first)
+                    myPageViewModel.isSelectSixthReason.value = false
+                }
+                R.id.unregister_reason_second -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_second)
+                    myPageViewModel.isSelectSixthReason.value = false
+                }
+                R.id.unregister_reason_third -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_third)
+                    myPageViewModel.isSelectSixthReason.value = false
+                }
+                R.id.unregister_reason_fourth -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_fourth)
+                    myPageViewModel.isSelectSixthReason.value = false
+                }
+                R.id.unregister_reason_fifth -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_fifth)
+                    myPageViewModel.isSelectSixthReason.value = false
+                }
+                R.id.unregister_reason_sixth -> {
+                    binding.tvChooseReason.setText(R.string.unregister_reason_sixth)
+                    myPageViewModel.isSelectSixthReason.value = true
+                }
+            }
```

Comment:
지금 당장 고쳐보자면 일단은 이렇습니다..!! 다른 방법으로 최적화 하는게 더 좋을 것 같네용!
```suggestion
myPageViewModel.isSelectSixthReason.value = checkedID == R.id.unregister_reason_sixth
binding.tvChooseReason.text = when (checkedID) {
    R.id.unregister_reason_first -> {
        getString(R.string.unregister_reason_first)
    }
    R.id.unregister_reason_second -> {
        getString(R.string.unregister_reason_second)
    }
    R.id.unregister_reason_third -> {
        getString(R.string.unregister_reason_third)
    }
    R.id.unregister_reason_fourth -> {
        getString(R.string.unregister_reason_fourth)
    }
    R.id.unregister_reason_fifth -> {
        getString(R.string.unregister_reason_fifth)
    }
    R.id.unregister_reason_sixth -> {
        getString(R.string.unregister_reason_sixth)
    }
    else -> {
        ""
    }
}
```

Reply:
이대로 BindingAdapter 로 빼버리는 것두 깔끔할듯 !!

Reply:
오호 그런방법이...담에 리팩토링 때 다시 해보겠습니당!

Reply:
근데 정렬 안된 상태로 들어가긴할거에요 ㅎ,,ㅎ

Reply:
오헐 신기하네용 ㅎㅎ 감사합니당!